File: Python3/674. Longest Continuous Increasing Subsequence.py
# 72ms, 98.90%; 13.8 MB, 100.00%
class Solution:
    def findLengthOfLCIS(self, nums: list) -> int:
        max_len = 0
        cur_len, last = 0, -1
        for i in nums:
            if i > last:
                cur_len += 1
                max_len = max(max_len, cur_len)
            else:
                cur_len = 1
            last = i
        return max(max_len, cur_len)

File: Cpp/674. Longest Continuous Increasing Subsequence.cpp
// 16ms, 56.70%; 9.3MB, 94.44%
class Solution
{
public:
    int findLengthOfLCIS(const vector<int> &nums)
    {
        int res = 0;
        int curLen = 0, last = -1;
        for (auto n : nums) {
            if (n > last) {
                curLen += 1;
                res = max(res, curLen);
            } else {
                curLen = 1;
            }
            last = n;
        }
        return max(res, curLen);
    }
};
